spark/src/main/java/org/apache/iceberg/spark/source/BaseDataReader.java

```diff
@@ -56,6 +56,7 @@
 
   private CloseableIterator<T> currentIterator;
   private T current = null;
+  private FileScanTask currentTask = null;
```

Comment:
Did you intend to set this in the constructor?

Reply:
Not really, does it matter if it is in the constructor? It is going to be set to null in the constructor anyway. Similar to the `current` variable above it.

Reply:
Sorry, I missed that this is getting set in `next`. I thought that change was just to indentation. Nevermind!